rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/delay/TbMsgDelayNode.java

```diff
@@ -89,17 +96,27 @@ public void onMsg(TbContext ctx, TbMsg msg) {
     }
 
     private long getDelay(TbMsg msg) {
-        int periodInSeconds;
-        if (config.isUseMetadataPeriodInSecondsPatterns()) {
-            if (isParsable(msg, config.getPeriodInSecondsPattern())) {
-                periodInSeconds = Integer.parseInt(TbNodeUtils.processPattern(config.getPeriodInSecondsPattern(), msg));
+        int periodValue;
+        if (config.isUsePeriodValuePattern()) {
+            if (isParsable(msg, config.getPeriodValuePattern())) {
+                periodValue = Integer.parseInt(TbNodeUtils.processPattern(config.getPeriodValuePattern(), msg));
             } else {
-                throw new RuntimeException("Can't parse period in seconds from metadata using pattern: " + config.getPeriodInSecondsPattern());
+                throw new RuntimeException("Can't parse period value using pattern: " + config.getPeriodValuePattern());
+            }
+        } else {
+            periodValue = config.getPeriodValue();
+        }
+        TimeUnit periodTimeUnit;
+        if (config.isUsePeriodTimeUnitPattern()) {
+            try {
+                periodTimeUnit = TimeUnit.valueOf(TbNodeUtils.processPattern(config.getPeriodTimeUnitPattern(), msg));
+            } catch (IllegalArgumentException | NullPointerException e) {
+                throw new RuntimeException("Can't parse period time unit using pattern: " + config.getPeriodTimeUnitPattern());
             }
         } else {
-            periodInSeconds = config.getPeriodInSeconds();
+            periodTimeUnit = config.getPeriodTimeUnit();
         }
-        return TimeUnit.SECONDS.toMillis(periodInSeconds);
+        return periodTimeUnit.toMillis(periodValue);
     }
 
     private boolean isParsable(TbMsg msg, String pattern) {
```

Comment:
Consider to remove method `isParsable` and use a try-catch block for the periodValuePattern in the same way as you did for the periodTimeUnitPattern